feature-libs/product-configurator/rulebased/components/attribute/composition/configurator-attribute-composition.directive.ts

```diff
@@ -10,34 +10,49 @@ import {
   Injector,
   Input,
   isDevMode,
-  OnInit,
+  OnChanges,
   ViewContainerRef,
 } from '@angular/core';
-import { LoggerService } from '@spartacus/core';
+import { LoggerService, ObjectComparisonUtils } from '@spartacus/core';
 import { ConfiguratorAttributeCompositionConfig } from './configurator-attribute-composition.config';
 import { ConfiguratorAttributeCompositionContext } from './configurator-attribute-composition.model';
+import { Configurator } from '@spartacus/product-configurator/rulebased';
 
 @Directive({
   selector: '[cxConfiguratorAttributeComponent]',
 })
-export class ConfiguratorAttributeCompositionDirective implements OnInit {
+export class ConfiguratorAttributeCompositionDirective implements OnChanges {
   @Input('cxConfiguratorAttributeComponent')
   context: ConfiguratorAttributeCompositionContext;
 
   protected logger = inject(LoggerService);
+  protected attribute: Configurator.Attribute;
 
   constructor(
     protected vcr: ViewContainerRef,
     protected configuratorAttributeCompositionConfig: ConfiguratorAttributeCompositionConfig
   ) {}
 
-  ngOnInit(): void {
-    const componentKey = this.context.componentKey;
```

Comment:
Removing ngOnInit and moveing the whole logic to ngOnChanges is a breaking change. Customers may have relied on `ngOnInit` being implemented, e.g. they could have called in their customized component `super.ngOnInit()`.
Therefore:
- let's leave ngOnInit, but wrap its body with a feature toggle
- let's leave the added ngOnChanges, but wrap it's body with an opposite feature toggle

Reply:
just skipped the whole FT part for the POC, but we will do it for the productive solution - but could to have this comment as reminder